### ISSUE TITLE
v14.2 causing build error for Material UI

### ISSUE THREAD

**Issue by nunesunil** [vercel/next.js#64369]
### Link to the code that reproduces this issue

https://github.com/nunesunil/next-js

### To Reproduce

Start the appliction for creating build(npm run build), then the build fails with the following error -

sunil@Sunils-MBP-M2 my-app % npm run build

> my-app@0.1.0 build
> next build

  ▲ Next.js 14.2.0

   Creating an optimized production build ...
 ✓ Compiled successfully
 ✓ Linting and checking validity of types    
 ✓ Collecting page data    
   Generating static pages (0/5)  [    ]Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object.
    at nM (/Users/sunil/Desktop/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:61599)
    at nM (/Users/sunil/Desktop/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:61546)
    at nN (/Users/sunil/Desktop/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64546)
    at nB (/Users/sunil/Desktop/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:67538)
    at nM (/Users/sunil/Desktop/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:58560)
    at nN (/Users/sunil/Desktop/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64546)
    at nB (/Users/sunil/Desktop/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:67538)
    at nD (/Users/sunil/Desktop/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:66680)
    at nN (/Users/sunil/Desktop/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64853)
    at nB (/Users/sunil/Desktop/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:67538) {
  digest: '4272938947'
}

Error occurred prerendering page "/". Read more: https://nextjs.org/docs/messages/prerender-error

Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object.
    at nM (/Users/sunil/Desktop/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:61599)
    at nM (/Users/sunil/Desktop/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:61546)
    at nN (/Users/sunil/Desktop/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64546)
    at nB (/Users/sunil/Desktop/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:67538)
    at nM (/Users/sunil/Desktop/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:58560)
    at nN (/Users/sunil/Desktop/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64546)
    at nB (/Users/sunil/Desktop/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:67538)
    at nD (/Users/sunil/Desktop/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:66680)
    at nN (/Users/sunil/Desktop/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64853)
    at nB (/Users/sunil/Desktop/my-app/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:67538)
 ✓ Generating static pages (5/5)

> Export encountered errors on following paths:
        /page: /
sunil@Sunils-MBP-M2 my-app % 
 

### Current vs. Expected behavior

The build is expected to create a static pages.

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.4.0: Fri Mar 15 00:12:49 PDT 2024; root:xnu-10063.101.17~1/RELEASE_ARM64_T6020
  Available memory (MB): 32768
  Available CPU cores: 10
Binaries:
  Node: 20.12.2
  npm: 10.5.2
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.2.0 // Latest available version is detected (14.2.0).
  eslint-config-next: N/A
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.4.5
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Not sure, App Router

### Which stage(s) are affected? (Select all that apply)

next build (local), Vercel (Deployed)

### Additional context

If I use h1 Tag instead of MUI Typography in page build is success.

<sub>[NEXT-3101](https://linear.app/vercel/issue/NEXT-3101/v142-causing-build-error-for-material-ui)</sub>

**Comment by nizioleque** [comment]
This is getting ridiculous. Every second Next.js update breaks compatibility with MUI. It was the same a couple versions ago. Do you guys really have no tests for that scenario - building a basic MUI app?

**Comment by pedroSoaresll** [comment]
I am facing the same here (again).

**Comment by Thinkscape** [comment]
My app compiles, but pages crash with:

```
 [Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object.] {
```

To repro have a `page.tsx` like so:
```tsx
import { Typography } from "@mui/material";

export function Page() {
  return <Typography>hello world</Typography>;
}

```

**Comment by Juneezee** [comment]
I got the same build error too with MUI 5.15.15.

I tested the canary versions since 14.1.0, and found that the last-known working canary is [14.1.1-canary.62](https://github.com/vercel/next.js/releases/tag/v14.1.1-canary.62). Starting from [14.1.1-canary.63](https://github.com/vercel/next.js/releases/tag/v14.1.1-canary.63), the build fails consistently with this **prerender error** all the way to [14.2.0](https://github.com/vercel/next.js/releases/tag/v14.2.0).

There must be something in between https://github.com/vercel/next.js/compare/v14.1.1-canary.62...v14.1.1-canary.63 that breaks the build.

**Comment by theexiile1305** [comment]
https://github.com/hm-edu/shibboleth-sp-oidc/pull/33

**Comment by 0xfima** [comment]
I got same error with @mui/icons-material: 5.15.14 and next: 14.2.0

Simple example of Next.Js page
```
import { Grid } from '@mui/material';

export default function Page() {
    return (
        <Grid container spacing={2}>
            Test
        </Grid>
}
```
After got same error:
`[=   ]Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object.`

**Comment by Jordaneisenburger** [comment]
Same for me after upgrading to 14.2.0 builds started failing for pages where we use `@mui/icons-material`

**Comment by manacy-keyvalue** [comment]
Experiencing the same issue after upgrading to v14.2.
The execution of `next build` runs successfully but the following error occurs whenever a page is loaded via `next start` with the usage of @mui/icons-material:
`Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object.
    at nM (/Users/manacy/Desktop/projects/AM-I/crm-web/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:61599)
    at nM (/Users/manacy/Desktop/projects/AM-I/crm-web/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:61546)
    at nN (/Users/manacy/Desktop/projects/AM-I/crm-web/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64546)
    at nN (/Users/manacy/Desktop/projects/AM-I/crm-web/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64825)
    at nB (/Users/manacy/Desktop/projects/AM-I/crm-web/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:67538)
    at nD (/Users/manacy/Desktop/projects/AM-I/crm-web/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:66680)
    at nN (/Users/manacy/Desktop/projects/AM-I/crm-web/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64853)
    at nB (/Users/manacy/Desktop/projects/AM-I/crm-web/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:67538)
    at nD (/Users/manacy/Desktop/projects/AM-I/crm-web/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:66680)
    at nN (/Users/manacy/Desktop/projects/AM-I/crm-web/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64853) {
  digest: '2890794125'
}
`
For pages that don't use @mui/icons-material, there is no error so far.

Also, not able to reproduce this issue in my local (run with `next dev`). It only occurs in the build with `next start`. There should be a consistent behaviour across both local development and builds.

**Comment by huozhi** [comment]
Downgrading mui to 5.15.6 works for me with the reproduction (next 14.2), it starts breaking since 5.15.7 with next 14.2

**Comment by subvertallchris** [comment]
I'm receiving this same error intermittently in prod after upgrading to 14.2.0. I can't say for sure it's from MUI but I am using MUI 5.15.3. Some symptoms different: builds fine, most things work fine, but some navigations between pages (esp if I load a particular page and click a particular Link on page) triggers it a good 90% of the time.

**Comment by melihplt** [comment]
In my case, it was Mui Divider component. Wrapping it with a 'use client' component solved the problem.

**Comment by huozhi** [comment]
Preparing a fix in https://github.com/vercel/next.js/issues/64369

**Comment by subvertallchris** [comment]
@huozhi Do you expect this to also resolve intermittent appearances of this error that appeared in production builds as described [here](https://github.com/vercel/next.js/issues/64369#issuecomment-2052627491)?

**Comment by lucasvieirasilva** [comment]
@Jordaneisenburger same here, even after the https://github.com/vercel/next.js/pull/64467 fix, I've opened a new issue about it: https://github.com/vercel/next.js/issues/64518

**Comment by nunesunil** [comment]
> Preparing a fix in #64467

Works if "import { Typography } from '@mui/material'" but not working if  "import Typography from '@mui/material/Typography'".

**Comment by huozhi** [comment]
@nunesunil Please check the update in https://github.com/vercel/next.js/issues/64518#issuecomment-2063478810

**Comment by huozhi** [comment]
The fix is published in 14.2.3, please upgrade to the new version 🙏
